### PR TITLE
[ERE-1826] Implement e2e feature tests

### PR DESCRIPTION
**Pull request by ppettitau**
Added view for local in-memory Django mail outbox to easily check and clear emails for testing
Added tests for:
* patient self registration and activation
* user password management (forgot and reset)
* Enable/Disable Two-factor authentication

### CODE REVIEW

File: rdrf/rdrf/apps.py [hunk 1]
```diff
@@ -12,8 +12,6 @@ class RDRFConfig(AppConfig):
 
     def ready(self):
         logger.info("running RDRFConfig.ready ... ")
-        import rdrf.account_handling.backends
```
Comment (ppettitau) [inline rdrf/rdrf/apps.py]: Should I put these back in? not really sure what they're there for...

Comment (ScriptSmith) [reply rdrf/rdrf/apps.py]: Yeah I think so, I believe they are setting up [signals](https://docs.djangoproject.com/en/dev/topics/signals/)

Comment (ppettitau) [reply rdrf/rdrf/apps.py]: Done.

File: rdrf/rdrf/testing/behaviour/features/auth/steps.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    captcha_iframe_element = world.browser.find_element_by_xpath(
+        "//iframe[@role='presentation']")
+
+    world.browser.switch_to.frame(captcha_iframe_element)
+    scroll_to_y(500)
+
+    world.browser.find_element_by_id('recaptcha-anchor').send_keys(Keys.SPACE)
+
+    time.sleep(4)
```
Comment (ScriptSmith) [inline rdrf/rdrf/testing/behaviour/features/auth/steps.py]: I think it'd be better to wait for the `registration-submit` button to be enabled by `reCaptchaCallback`. Sleeping for a set amount of time can cause the test to fail when a related change alters the timing but there hasn't actually been a regression.

File: rdrf/rdrf/templates/debug/outbox.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{% extends "rdrf_cdes/base.html" %}
+
+{% load i18n %}
+
+{% block content %}
+  <h1>Mail Outbox</h1>
+  {% if mail_messages %}
+    <table class="table table-striped" id="id_messages">
```
Comment (ScriptSmith) [inline rdrf/rdrf/templates/debug/outbox.html:8]: nit: Long messages make the table columns spill outside the page. Could either make the text wrap, or the individual cells have an `overflow` css property
image

**Review by ScriptSmith — CHANGES_REQUESTED** [review]
Great stuff.Very glad to finally have this :smile:

File: rdrf/rdrf/testing/behaviour/features/auth/steps.py (hunk 2 above)
Comment (ppettitau) [reply rdrf/rdrf/testing/behaviour/features/auth/steps.py]: This was existing (unused) code that I moved over, but I can have a look.

**Review by ScriptSmith — APPROVED** [review]

**Pull request merged**